test/test_turbulence_closures.jl

```diff
@@ -148,6 +156,45 @@ function time_step_with_variable_discrete_diffusivity(arch)
     return true
 end
 
+function diffusivity_fields_sizes_are_correct(arch)
+    grid = RectilinearGrid(arch, size=(2, 3, 4), extent=(1, 2, 3))
+
+    closure = Smagorinsky(coefficient=DynamicCoefficient(averaging=1))
+    model = NonhydrostaticModel(; grid, closure)
+    @test size(model.diffusivity_fields.𝒥ᴸᴹ) == (1, grid.Ny, grid.Nz)
+    @test size(model.diffusivity_fields.𝒥ᴹᴹ) == (1, grid.Ny, grid.Nz)
+    @test size(model.diffusivity_fields.LM)  == size(grid)
+    @test size(model.diffusivity_fields.MM)  == size(grid)
+    @test size(model.diffusivity_fields.Σ)   == size(grid)
+    @test size(model.diffusivity_fields.Σ̄)   == size(grid)
+
+    closure = DynamicSmagorinsky(averaging=(1, 2))
+    model = NonhydrostaticModel(; grid, closure)
+    @test size(model.diffusivity_fields.𝒥ᴸᴹ) == (1, 1, grid.Nz)
+    @test size(model.diffusivity_fields.𝒥ᴹᴹ) == (1, 1, grid.Nz)
+
+    closure = DynamicSmagorinsky(averaging=Colon())
+    model = NonhydrostaticModel(; grid, closure)
+    @test size(model.diffusivity_fields.𝒥ᴸᴹ) == (1, 1, 1)
+    @test size(model.diffusivity_fields.𝒥ᴹᴹ) == (1, 1, 1)
+
+    closure = DynamicSmagorinsky(averaging=(2, 3))
+    model = NonhydrostaticModel(; grid, closure)
+    @test size(model.diffusivity_fields.𝒥ᴸᴹ) == (grid.Nx, 1, 1)
+    @test size(model.diffusivity_fields.𝒥ᴹᴹ) == (grid.Nx, 1, 1)
+
+    closure = DynamicSmagorinsky(averaging=LagrangianAveraging())
+    model = NonhydrostaticModel(; grid, closure)
+    @test size(model.diffusivity_fields.𝒥ᴸᴹ)  == size(grid)
+    @test size(model.diffusivity_fields.𝒥ᴹᴹ)  == size(grid)
+    @test size(model.diffusivity_fields.𝒥ᴸᴹ⁻) == size(grid)
+    @test size(model.diffusivity_fields.𝒥ᴹᴹ⁻) == size(grid)
+    @test size(model.diffusivity_fields.Σ)    == size(grid)
+    @test size(model.diffusivity_fields.Σ̄)    == size(grid)
```

Comment:
What's the purpose of all of these tests? Are they looking for a specific error that cropped up?

Reply:
No, I just thought it was helpful to test that the diffusivity fields were the appropriate size given that the directionally-averaged versions have sizes that are different from the underlying `grid`.